### PR TITLE
Record routines in query event

### PR DESCRIPTION
**Pull request by martint**

### CODE REVIEW

File: presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java [hunk 1]
```diff
@@ -276,6 +277,13 @@ public void testReferencedTables()
         ColumnInfo column = table.getColumns().get(0);
         assertEquals(column.getColumn(), "linenumber");
         assertTrue(column.getMasks().isEmpty());
+
+        List<RoutineInfo> routines = event.getMetadata().getRoutines();
+        assertEquals(tables.size(), 1);
+
+        RoutineInfo routine = routines.get(0);
+        assertEquals(routine.getRoutine(), "sum");
+        assertEquals(routine.getAuthorization(), "user");
```
Comment (kokosing) [inline presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java]: Can you please also add a test for procedure?
Also we have two access checks: one for function and one for procedure. So we need to be able to recognize if the routine is a function or procedure.

Comment (martint) [reply presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java]: > Also we have two access checks: one for function and one for procedure. So we need to be able to recognize if the routine is a function or procedure.

Why does it matter?

Comment (martint) [reply presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java]: > Can you please also add a test for procedure?

Unfortunately, we don't have any infrastructure to test events for non-SQL tasks or register procedures with the testing query runners.

Comment (kokosing) [reply presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java]: > Why does it matter?

We have two access control checks. Same for tables and views. However we have single table info entity to handle tables and views. So let's leave it as is for now.

**Review by kokosing — APPROVED** [review]

**Pull request merged**